common/src/main/java/com/wynntils/features/DiscordRichPresenceFeature.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features;
+
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.mc.event.ConnectionEvent;
+import com.wynntils.mc.event.PlayerMoveEvent;
+import com.wynntils.mc.event.PlayerTeleportEvent;
+import com.wynntils.mc.event.SetXpEvent;
+import com.wynntils.models.character.event.CharacterUpdateEvent;
+import com.wynntils.models.territories.profile.TerritoryProfile;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import com.wynntils.models.worlds.type.WorldState;
+import com.wynntils.utils.mc.McUtils;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class DiscordRichPresenceFeature extends Feature {
+
+    @RegisterConfig
+    public final Config<Boolean> displayLocation = new Config<>(true);
+
+    @RegisterConfig
+    public final Config<Boolean> displayCharacterInfo = new Config<>(true);
+
+    @RegisterConfig
+    public final Config<Boolean> displayWorld = new Config<>(true);
+
+    private TerritoryProfile lastTerritoryProfile = null;
+
+    private void handleLocationChange() {
+        if (displayLocation.get()) {
+            TerritoryProfile territoryProfile = Models.Territory.getTerritoryProfileForPosition(
+                    McUtils.player().position());
+            if (territoryProfile == null || territoryProfile == lastTerritoryProfile) return;
+            lastTerritoryProfile = territoryProfile;
+            String location = territoryProfile.getName();
+            Managers.Discord.setLocation(location);
+        } else {
+            Managers.Discord.setLocation("");
+            lastTerritoryProfile = null;
+        }
+    }
+
+    @SubscribeEvent
+    public void onPlayerMove(PlayerMoveEvent event) {
+        handleLocationChange();
+    }
+
+    @SubscribeEvent
+    public void onPlayerTeleport(PlayerTeleportEvent event) {
+        handleLocationChange();
+    }
+
+    @SubscribeEvent
+    public void onCharacterUpdate(CharacterUpdateEvent event) {
+        if (displayCharacterInfo.get()) {
+            Managers.Discord.setClassType(Models.Character.getClassType());
```

Comment:
IMO this is a little weird connection between managers and functions, perhaps @magicus can suggest a better way when he gets to review this.

Reply:
You mean between managers and features, I presume. 

Well, maybe yes. I think there are two reasonable splits to make here, and the current solutions blends those together.

Option 1 is to have a DiscordManager/DiscordExternalModel (or whatever we should call it) that is basically a thin wrapper around the discord API. In that case, it should not know about concepts like "Location" but rather "Large Text" (or whatever it was called).

Option 2 is to have the DiscordManager/DiscordExternalModel specify in detail how we are allowed to set rich presence info on discord, and deal in terms of "Location" and "Class" etc. But in that case, the model should itself listen to the events needed to keep it up to date, which renders the Feature almost moot. At this point, it would only be there to store config and have a way to turn this on/off; since we can't do that for Models yet.

I'm not sure which I prefer. But I think Option 1 have some kind of appeal. That would make the DiscordManager expose a more high level API with text and images (I guess?) to display, and the responsibility of setting those to sane Wynncraft values should fall on this feature. I think that would be neat.